pkg/agent/controller/noderoute/node_route_controller_test.go

```diff
@@ -445,3 +476,109 @@ func TestCreateIPSecTunnelPortCert(t *testing.T) {
 		})
 	}
 }
+
+func TestGetNodeMAC(t *testing.T) {
+	mac := "00:1B:44:11:3A:B7"
+	expectedMac, err := net.ParseMAC(mac)
+	require.NoError(t, err)
+	node1 := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{types.NodeMACAddressAnnotationKey: mac},
+		},
+	}
+
+	got, err := getNodeMAC(node1)
+	require.NoError(t, err)
+	assert.Equal(t, expectedMac, got)
+}
+
+func TestParseTunnelInterfaceConfig(t *testing.T) {
+	tests := []struct {
+		name                    string
+		portData                *ovsconfig.OVSPortData
+		portConfig              *interfacestore.OVSPortConfig
+		expectedInterfaceConfig *interfacestore.InterfaceConfig
+	}{
+		{
+			name: "Tunnel interface",
+			portData: &ovsconfig.OVSPortData{
+				Name:   "antrea-tun0",
+				IFType: "gre",
+				Options: map[string]string{
+					"dst_port": "2",
+				},
+				OFPort: 1,
+			},
+			portConfig: &interfacestore.OVSPortConfig{
+				OFPort: 1,
+			},
+			expectedInterfaceConfig: &interfacestore.InterfaceConfig{
+				InterfaceName: "antrea-tun0",
+				Type:          interfacestore.TunnelInterface,
+				TunnelInterfaceConfig: &interfacestore.TunnelInterfaceConfig{
+					Type:            ovsconfig.TunnelType("gre"),
+					DestinationPort: 2,
+				},
+				OVSPortConfig: &interfacestore.OVSPortConfig{OFPort: 1}},
+		},
+		{
+			name: "IPSec tunnel interface",
+			portData: &ovsconfig.OVSPortData{
+				Name:   "antrea-ipsec-tun",
+				IFType: "gre",
+				Options: map[string]string{
+					"remote_name": "testNode",
+					"dst_port":    "2",
+				},
+				OFPort:      1,
+				ExternalIDs: map[string]string{ovsExternalIDNodeName: "testNode"},
+			},
+			portConfig: &interfacestore.OVSPortConfig{
+				OFPort: 1,
+			},
+			expectedInterfaceConfig: &interfacestore.InterfaceConfig{
+				InterfaceName: "antrea-ipsec-tun",
+				Type:          interfacestore.IPSecTunnelInterface,
+				TunnelInterfaceConfig: &interfacestore.TunnelInterfaceConfig{
+					Type:       ovsconfig.TunnelType("gre"),
+					NodeName:   "testNode",
+					RemoteName: "testNode",
+				},
+				OVSPortConfig: &interfacestore.OVSPortConfig{OFPort: 1}},
+		},
```

Comment:
nit: Can add a case for portData with no options

Reply:
added.